### ISSUE TITLE
ffmpeg-based alternative for MgObject.motiongrams() and motionvideo()

### ISSUE THREAD

**Issue by balintlaczko** [fourMs/MGT-python#153]
Back when every major function got an ffmpeg-based backend, I decided to keep `motion()` as it is, since exchanging matrices frame-by-frame (which is necessary for the plotting and csv generation) would anyway create the same kind of bottleneck we have right now. But then I made the shortcuts based on `motion()` such as `motiondata()`, `motionvideo()` and `motiongrams()` which basically call `motion()` set up to only render/output one thing (the data, video or motiongrams, respectively).

Meanwhile ffmpeg finally added the `medianfilter` to the public build. This means I can recreate the exact same pipeline we use with OpenCV. So the proposal here is to change `motionvideo()` and `motiongrams()` to use an ffmpeg-backend instead, so we get the big speed upgrade, without losing integrity/quality of the product.

**Comment by alexarje** [comment]
Yes, sounds like a good idea!

**Comment by balintlaczko** [comment]
This is now implemented. I realized that the median filter in ffmpeg is somehow(?) stronger than `scipy.signal.medfilt2d`. It also makes the rendering much slower. So I added a `use_median` parameter which is `False` by default. I also added a hidden "switch" which disables the median filter if `filtertype='blob'`. Since erosion selects the minimum of the kernel already, applying a median filter on what's already eroded with the same kernel size seems to me like an expensive no-op.

Of course there is a huge speed upgrade with using ffmpeg but I have to underline this even more in case of the motiongrams. What earlier took like a minute with `mg_motion` (set to only output motiongrams) takes around one second(!) with `mg_motiongrams` (which now uses the new `_utils.motiongrams_ffmpeg`).
I don't know what they are doing at ffmpeg, but they are definitely doing it.
